signer/src/domain.rs

```diff
@@ -0,0 +1,134 @@
+//! Signer domain and helpers
+//!
+//! Shorthands and helpers for base and scalar field elements
+
+use ark_ec::AffineCurve;
+use ark_ff::PrimeField; // for into_repr()
+
+use mina_curves::pasta::pallas as Pallas;
+
+/// Affine curve point type
+pub use Pallas::Affine as CurvePoint;
+/// Base field element type
+pub type BaseField = <CurvePoint as AffineCurve>::BaseField;
+/// Scalar field element type
+pub type ScalarField = <CurvePoint as AffineCurve>::ScalarField;
+
+use ark_serialize::CanonicalSerialize;
+
+/// Field element helpers
+pub trait FieldHelpers<F: PrimeField> {
```

Comment:
can you move this to o1-utils?

Reply:
Yes, I plan to do this in this ticket https://github.com/o1-labs/proof-systems/issues/242